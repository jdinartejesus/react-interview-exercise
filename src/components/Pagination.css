.pagination {
  margin: 0 auto;
  padding: 10px;
}

:local(.Pagination) {
  background-color: white;
  display: flex;
  align-items: center;
}
