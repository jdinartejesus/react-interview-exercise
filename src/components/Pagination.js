import React, { Component, PropTypes } from 'react';
import styles from './Pagination.css';
import { times } from 'lodash';

class Pagination extends Component {
  render () {
    const { totalPages, onHandlerPrevious, onHandlerNext, onHandlePageNumber } = this.props;

    return (
      <div className={styles.Pagination}>
        <ul className="pagination">
          <li>
            <a href="#" className="btn btn-default" onClick={onHandlerPrevious}>
              <span aria-hidden="true">&laquo;</span>
            </a>
          </li>
          {
            times(totalPages, (i) => (
              <li>
                <a href="#" className="btn btn-default" onClick={onHandlePageNumber.bind(null, i)} key={i} >
                  {i}
                </a>
              </li>
            ))
          }
          <li>
            <a href="#" className="btn btn-default" onClick={onHandlerNext}>
              <span aria-hidden="true">&raquo;</span>
            </a>
          </li>
        </ul>
      </div>
    )
  }
}

const { number, func } = PropTypes

Pagination.propTypes = {
  totalPages: number.isRequired,
  onHandlerPrevious: func.isRequired,
  onHandlerNext: func.isRequired,
  onHandlePageNumber: func.isRequired
}

export default Pagination
