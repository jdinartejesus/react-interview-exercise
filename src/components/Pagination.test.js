import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Pagination from './Pagination';

describe('components/Pagination', () => {
  const defaultTotalPages = 5;
  const onHandlerNext = (number) => ( number )
  const onHandlerPrevious = (number) => ( number )
  const onHandlePageNumber = (number) => ( number )

  it('should match it\'s empty snapshot', () => {
    const tree = renderer.create(
      <Pagination
        totalPages={defaultTotalPages}
        onHandlerNext={onHandlerNext}
        onHandlerPrevious={onHandlerPrevious}
        onHandlePageNumber={onHandlePageNumber}
      />
    )

    expect(tree).toMatchSnapshot();
  })

  it('should display of buttons for given TotalPages', () => {
    const component = shallow(
      <Pagination
        totalPages={defaultTotalPages}
        onHandlerNext={onHandlerNext}
        onHandlerPrevious={onHandlerPrevious}
        onHandlePageNumber={onHandlePageNumber}
      />
    );

    expect(component.find('a').not('a[aria-label]')).toHaveLength(defaultTotalPages);
  })

  it('should return next page number', () => {
    const onHandlerNext = jest.fn();
    const component = shallow(
      <Pagination
        totalPages={defaultTotalPages}
        onHandlerNext={onHandlerNext}
        onHandlerPrevious={onHandlerPrevious}
        onHandlePageNumber={onHandlePageNumber}
      />
    );

    component.find('a[aria-label="Next"]').simulate('click');
    expect(onHandlerNext).toHaveBeenCalled();
  })

  it('should return previous page number', () => {
    const onHandlerPrevious = jest.fn();
    const component = shallow(
      <Pagination
        totalPages={defaultTotalPages}
        onHandlerNext={onHandlerNext}
        onHandlerPrevious={onHandlerPrevious}
        onHandlePageNumber={onHandlePageNumber}
      />
    );

    component.find('a[aria-label="Previous"]').simulate('click');
    expect(onHandlerPrevious).toHaveBeenCalled();
  })

  it('should return clicked page number', () => {
    const onHandlePageNumber = jest.fn();
    const component = shallow(
      <Pagination
        totalPages={defaultTotalPages}
        onHandlerNext={onHandlerNext}
        onHandlerPrevious={onHandlerPrevious}
        onHandlePageNumber={onHandlePageNumber}
      />
    );

    component.find('a').not('a[aria-label]').first().simulate('click')
    expect(onHandlePageNumber).toBeCalledWith(0);
  })
})
