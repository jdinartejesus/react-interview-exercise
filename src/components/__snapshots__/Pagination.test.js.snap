exports[`components/Pagination should match it's empty snapshot 1`] = `
<div
  className={undefined}>
  <ul
    className="pagination">
    <li>
      <a
        aria-label="Previous"
        className="btn btn-default"
        href="#"
        onClick={[Function]}>
        <span
          aria-hidden="true">
          «
        </span>
      </a>
    </li>
    <li>
      <a
        className="btn btn-default"
        href="#"
        onClick={[Function]}>
        1
      </a>
    </li>
    <li>
      <a
        className="btn btn-default"
        href="#"
        onClick={[Function]}>
        2
      </a>
    </li>
    <li>
      <a
        className="btn btn-default"
        href="#"
        onClick={[Function]}>
        3
      </a>
    </li>
    <li>
      <a
        className="btn btn-default"
        href="#"
        onClick={[Function]}>
        4
      </a>
    </li>
    <li>
      <a
        className="btn btn-default"
        href="#"
        onClick={[Function]}>
        5
      </a>
    </li>
    <li>
      <a
        aria-label="Next"
        className="btn btn-default"
        href="#"
        onClick={[Function]}>
        <span
          aria-hidden="true">
          »
        </span>
      </a>
    </li>
  </ul>
</div>
`;
